rts/motoko-rts/src/gc/incremental/partitioned_heap.rs

```diff
@@ -515,26 +701,40 @@ impl PartitionedHeap {
         }
     }
 
-    fn rank_partitions_by_garbage(&self) -> [usize; MAX_PARTITIONS] {
-        let mut ranked_partitions: [usize; MAX_PARTITIONS] = from_fn(|index| index);
-        sort(&mut ranked_partitions, &|left, right| {
-            self.get_partition(left)
-                .garbage_amount()
-                .cmp(&self.get_partition(right).garbage_amount())
-                .reverse()
-        });
+    unsafe fn temporary_array<M: Memory>(mem: &mut M, length: usize) -> *mut usize {
+        // No post allocation barrier as this RTS-internal blob can be collected by the GC.
```

Comment:
I guess you need to use a blob because Rust doesn't let you stack-allocate something of dynamic size `length`?

Reply:
Yes, unfortunately.